tools/clang/lib/Parse/ParseStmt.cpp

```diff
@@ -189,11 +189,11 @@ Parser::ParseStatementOrDeclarationAfterAttributes(StmtVector &Stmts,
     // punctuator, then we are using them as identifers. Need to change
     // the token type to tok::identifier and fall through to the next case.
     // E.g., center = <RHS>.
-    if (tok::isPunctuator(NextToken().getKind())) {
+    if (!tok::isPunctuator(NextToken().getKind())) {
+      goto tok_default_case;
+    } else {
```

Comment:
The goto makes this so weird. I'm guessing the problem here was that the fallthrough wasn't at the end?

Reply:
Yes.